LibreNMS/OS.php

```diff
@@ -130,21 +130,22 @@ public function getCacheByIndex($oid, $mib = null, $snmpflags = '-OQUs')
      * DO NOT use numeric oids with this function! The snmp result must contain only one oid.
      *
      * @param string $oid textual oid
-     * @param string $mib mib for this oid
+     * @param string $mib mib for this oid (optional)
+     * @param string $depth depth for snmpwalk_group (optional)
      * @return array array indexed by the snmp index with the value as the data returned by snmp
      */
-    public function getCacheTable($oid, $mib = null)
+    public function getCacheTable($oid, $mib = null, $depth = null)
     {
         if (str_contains($oid, '.')) {
             echo "Error: don't use this with numeric oids!\n";
             return null;
         }
 
         if (!isset($this->cache[$oid])) {
```

Comment:
This isset doesn't match the lines below.
You shouldn't need $depth as a key right?  Snmp tables have a fixed number of items in the index of a table.

Reply:
Unless you can be sure that nobody will ever call the function twice with different $depth ... 
The point here is that for complex tables, you may have a situation where you want to see them with different values at different places of the code (to loop over it in a different way). Having this cached without $depth as a key, means that a piece of code my ask for a depth and receive a cached value with another depth (and fail ...). 
I agree this is probably not so often, but it would be fairly difficult to troubleshoot. As a security and for coherence of the result of the cached value compared to the "non cached" equivalent, a nested hastable does not cost that much CPU (LibreNMS waits for IO 99% of the time (Network IO, Disk IO ...)).

Nice catch for the isset, missed it ! :(

Reply:
Depth is dictated by the how many values are in the index of the snmp table (you can see this in the mib).  This really can't change.

Reply:
In a perfectly defined mib, why not, but in real life, you sometimes try multiple depths in order to make your life easier in the loop.
Caching the depths values acts as a check. The same reason we check parameters in a function call. Useless until somebody calls the function with the wrong parameters :)
$depth as key can't be wrong and does not cost much of memory/performance. Not caching it exposes a risk if the function is misused (on purpose or by mistake). 

If you prefer that I remove it, I will of course.